core/site_scons/tools.py

```diff
@@ -140,3 +140,14 @@ def compress_action(target, source, env):
     )
 
     env.Depends(obj_program, compress)
+
+
+def embed_raw_binary(obj_program, env, section, target_, file):
+    obj_program.extend(
+        env.Command(
+            target=target_,
+            source=file,
+            action="$OBJCOPY -I binary -O elf32-littlearm -B arm"
+            f" --rename-section .data=.{section}" + " $SOURCE $TARGET",
```

Comment:
@TychoVrahe What's the purpose of renaming the `.data` section here?

Reply:
The objcopy input target is `-I binary`, so everything in that file is considered `.data`, so i need to differentiate it in linker script somehow to ensure proper placement. And also need to ensure that it isn't dropped by linker since it otherwise seems unused.

I could also do something like this in linker script
`    KEEP(build/firmware/build/kernel/kernel.o(.data));`
instead of current 
`    KEEP(*(.kernel));`
but if find current solution a bit nicer.

Reply:
Yes, thanks for the explanation. This solution is absolutely fine.